conditions.go

```diff
@@ -75,6 +75,60 @@ func (c Condition) Validate() error {
 	return nil
 }
 
+func (c Condition) MarshalJSON() ([]byte, error) {
+	var serialized string
+	if c != nil {
+		serialized = c.String()
+	}
+	return json.Marshal(fmt.Sprintf("cond:%s", serialized))
+}
+
+func (c *Condition) UnmarshalJSON(raw []byte) error {
+	var enc string
+	if err := json.Unmarshal(raw, &enc); err != nil {
+		return errors.Wrap(err, "cannot decode json")
+	}
+
+	// If the encoded string starts with a prefix, cut it off and use
+	// specified decoding method instead of default one.
+	chunks := strings.SplitN(enc, ":", 2)
+	format := chunks[0]
+	if len(chunks) == 1 {
+		format = "hex"
+	} else {
+		enc = chunks[1]
+	}
+
+	// No value zero the address.
+	if len(enc) == 0 {
+		*c = nil
+		return nil
+	}
+
+	switch format {
+	case "hex":
+		val, err := hex.DecodeString(enc)
+		if err != nil {
+			return errors.Wrap(err, "cannot decode hex")
+		}
+		*c = val
+		return nil
+	case "cond":
+		args := strings.Split(enc, "/")
```

Comment:
If you follow the above suggestion, we can reduce this function to simply this case. We have no prefix and just parse the condition info